UI: fix state mapping for Sample Changer widget

elmjag opened the pull request:
Treat disabled state as closed state in InOutSwitch component.
    
Change logic for handling DISABLED state in the InOutSwitch component. Make the DISABLED state equivalent to CLOSED state.
    
This solves the issue with the 'Sample Changer' InOutSwitch widget. Previously it was not possible to power on a disabled sample changer, as the command for DISABLED state was set to 'PowerOff'. Now when it is in DISABLED state, the command will be 'PowerOn'.
    
Note that for sample changers, the DISABLED state is used when they are in 'powered off' mode.

jbflo commented on ui/src/containers/BeamlineSetupContainer.jsx:
```diff
@@ -39,6 +39,25 @@ function BeamlineSetupContainer(props) {
 
+  function handleSendCommand(targetState) {
+    switch (targetState) {
+      case 'READY': {
+        sendCommand('PowerOn');
+        break;
+      }
```
I am Wondering , you send PowerOn when State is READY ?
I would have expert READY mean the Device is powered on , .

But good to know if it's the case

elmjag replied:
No, I send PowerOn command when I want to _switch_ to READY state. The switch case is on target state.

In practice I send PowerOn when in DISABLED state.

jbflo replied:
Ahh in that case you could  just change line 113 of the  InOutSwitch component to :
onClick={() => handleSwitch(openValue)}

and also line 115 to : 
Set: {openText}

That would be it .

elmjag replied:
Ah, yeah that would be better.

I'll test it, next week when I'm back from my vacation :)

elmjag replied:
After looking a bit closer at the InOutSwitch code, I think the right way to fix this is to treat 'DISABLED' state the same way as 'CLOSED' state.

jbflo commented on ui/src/containers/BeamlineSetupContainer.jsx:
```diff
@@ -261,11 +280,11 @@ function BeamlineSetupContainer(props) {
             <InOutSwitch
               openText="Power On"
               offText="Power Off"
-              openValue="PowerOn"
-              offValue="PowerOff"
+              openValue="READY"
+              offValue="DISABLED"
               labelText="Sample Changer"
               value={sampleChanger.state}
-              onSave={sendCommand}
+              onSave={handleSendCommand}
             />
```
I would suggest :

`            <InOutSwitch
              openText="Power On"
              offText="Power Off"
              openValue="PowerOff"
              offValue="PowerOn"
              labelText="Sample Changer"
              value={sampleChanger.state}
              onSave={sendCommand}            
/>
`
As you powerOn when state is READY
just  switch openValue to offValue would solve this,  
Maybe  ??

but I definitely think it is preferable not to have a another handle method handleSendCommand.
InOutSwitch already has a Handle method in.

elmjag replied:
I don't understand how that will work.

Sample Changers don't have PowerOff nor PowerOn states. The go between READ and DISABLED.

jbflo replied:
> I would suggest :
> 
> ` <InOutSwitch openText="Power On" offText="Power Off" openValue="PowerOff" offValue="PowerOn" labelText="Sample Changer" value={sampleChanger.state} onSave={sendCommand} />` As you powerOn when state is READY just switch openValue to offValue would solve this, Maybe ??
> 
> but I definitely think it is preferable not to have a another handle method handleSendCommand.
InOutSwitch already has a Handle method in.

Consider this as Not relevant:  see my other response on top,

Commit:
UI: treat disabled state as closed state in InOutSwitch

Change logic for handling DISABLED state in the InOutSwitch
component. Make the DISABLED state equivalent to CLOSED state.

This solves the issue with the 'Sample Changer' InOutSwitch widget.
Previously it was not possible to power on a disabled sample
changer, as the command for DISABLED state was set to 'PowerOff'.
Now when it is in DISABLED state, the command will be 'PowerOn'.

Note that for sample changers, the DISABLED state is used when
they are in 'powered off' mode.

elmjag commented:
After suggestions from @jbflo, I decided to take a different approach to fix the issue. This is more elegant way to deal with it. I have updated the description of this PR to describe the new approach.
